### PR TITLE
feat(image): adds method to create Associations from remote sources

### PR DESCRIPTION
**Pull request by jpower432**
Signed-off-by: Jennifer Power <barnabei.jennifer@gmail.com>

# Description

This PR adds methods to allow `AssociationSet` information to be gathered from local and remote sources.

Related to #302 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit tests added for remote layer association collection

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jpower432** [comment]
cc @afflom this is ready for logic review

**Review by afflom — APPROVED** [review]
I was able to test this with m2d and d2m workflows. It works as advertised. Over to @deejross for code review.

**Comment by jpower432** [comment]
@soltysh PTAL (priority #1). Thanks!

File: pkg/image/association.go [hunk 1]
```diff
@@ -219,146 +230,16 @@ func (as AssociationSet) validate() error {
...
-		return nil, fmt.Errorf("expected symlink or regular file mode, got: %b", m)
-	}
-	manifestBytes, err := ioutil.ReadFile(filepath.Clean(manifestPath))
-	if err != nil {
-		return nil, fmt.Errorf("error reading image manifest file: %v", err)
-	}
+func GetImageFromBlob(as AssociationSet, digest string) string {
+	for imageName, assocs := range as {
+		for _, assoc := range assocs {
+			for _, dgst := range assoc.LayerDigests {
```
Comment (soltysh) [inline pkg/image/association.go:236]: Why not use `AssociationSet#GetDigest` instead of manually iterating over `AssociaciationSet`'s?

**Review by soltysh — APPROVED** [review]
A couple of nits
/lgtm

**Pull request merged**